src/libs/antares/series/series.cpp

```diff
@@ -45,18 +46,37 @@ static bool checkAllElementsIdenticalOrOne(std::vector<uint> w)
     return std::adjacent_find(w.begin(), first_one, std::not_equal_to<uint>()) == first_one;
 }
 
-bool TimeSeries::Numbers::checkSeriesNumberOfColumnsConsistency() const
+static std::string errorMessage(const std::map<std::string, const TimeSeries*>& series)
```

Comment:
Ok. 
As, in the **std::map**'s values (reference to **TimeSeries**), we only use their number of columns, could we not directly have number of columns as values in this **std::map** ?
It would allow to use this boost join(...) algorithm